handler/src/main/java/io/netty/handler/timeout/IdleStateHandler.java

```diff
@@ -269,8 +269,10 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception
     @Override
     public void channelReadComplete(ChannelHandlerContext ctx) throws Exception {
         if (readerIdleTimeNanos > 0 || allIdleTimeNanos > 0) {
-            lastReadTime = System.nanoTime();
-            reading = false;
+            if (reading) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Merge this if statement with the enclosing one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1066)